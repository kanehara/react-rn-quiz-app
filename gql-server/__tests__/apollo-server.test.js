const { createTestClient } = require('apollo-server-testing')
const { server } = require('../apollo-server')
const nock = require('nock')

const MOCK_RESPONSE = {
  response_code: 0,
  results: [
    {
      category: "Entertainment: Video Games",
      type: "boolean",
      difficulty: "easy",
      question: "&quot;Undertale&quot; is an RPG created by Toby Fox and released in 2015.",
      correct_answer: "True",
      incorrect_answers: [
        "False"
      ]
    },
    {
      category: "Science: Mathematics",
      type: "multiple",
      difficulty: "hard",
      question: "How many zeptometres are inside one femtometre?",
      correct_answer: "1,000,000",
      incorrect_answers: [
        "10",
        "1,000,000,000",
        "1000"
      ]
    },
    {
      category: "Entertainment: Musicals & Theatres",
      type: "multiple",
      difficulty: "hard",
      question: "In Macbeth, the eyes of what animals were used in the Witches&#039; cauldron?",
      correct_answer: "Newts",
      incorrect_answers: [
        "Humans",
        "Sharks",
        "Squids"
      ]
    },
    {
      category: "General Knowledge",
      type: "multiple",
      difficulty: "hard",
      question: "Nephelococcygia is the practice of doing what?",
      correct_answer: "Finding shapes in clouds",
      incorrect_answers: [
        "Sleeping with your eyes open",
        "Breaking glass with your voice",
        "Swimming in freezing water"
      ]
    }
  ]
}

describe('gql server', () => {
  let query
  beforeEach(() => {
    const client = createTestClient(server)
    query = client.query
  })

  it('fetches questions', async () => {
    nock('https://opentdb.com')
    .get('/api.php?amount=10&difficulty=hard&type=boolean')
    .reply(200, MOCK_RESPONSE)
    const res = await query({ query: `
      query {
        questions {
          category
          type
          difficulty
          question
          correct_answer
          incorrect_answers
        }
      }
    ` })
    expect(res).toMatchSnapshot()
  })

  it('fetches questions with custom params', async () => {
    nock('https://opentdb.com')
    .get('/api.php?amount=1&difficulty=easy&type=custom')
    .reply(200, MOCK_RESPONSE)
    const res = await query({ query: `
      query {
        questions(amount:1, difficulty:"easy", type:"custom") {
          category
          type
          difficulty
          question
          correct_answer
          incorrect_answers
        }
      }
    ` })
    expect(res).toMatchSnapshot()
  })
})