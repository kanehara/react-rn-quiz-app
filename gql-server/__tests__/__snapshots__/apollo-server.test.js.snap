// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gql server fetches questions 1`] = `
Object {
  "data": Object {
    "questions": Array [
      Object {
        "category": "Entertainment: Video Games",
        "correct_answer": "True",
        "difficulty": "easy",
        "incorrect_answers": Array [
          "False",
        ],
        "question": "&quot;Undertale&quot; is an RPG created by Toby Fox and released in 2015.",
        "type": "boolean",
      },
      Object {
        "category": "Science: Mathematics",
        "correct_answer": "1,000,000",
        "difficulty": "hard",
        "incorrect_answers": Array [
          "10",
          "1,000,000,000",
          "1000",
        ],
        "question": "How many zeptometres are inside one femtometre?",
        "type": "multiple",
      },
      Object {
        "category": "Entertainment: Musicals & Theatres",
        "correct_answer": "Newts",
        "difficulty": "hard",
        "incorrect_answers": Array [
          "Humans",
          "Sharks",
          "Squids",
        ],
        "question": "In Macbeth, the eyes of what animals were used in the Witches&#039; cauldron?",
        "type": "multiple",
      },
      Object {
        "category": "General Knowledge",
        "correct_answer": "Finding shapes in clouds",
        "difficulty": "hard",
        "incorrect_answers": Array [
          "Sleeping with your eyes open",
          "Breaking glass with your voice",
          "Swimming in freezing water",
        ],
        "question": "Nephelococcygia is the practice of doing what?",
        "type": "multiple",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`gql server fetches questions with custom params 1`] = `
Object {
  "data": Object {
    "questions": Array [
      Object {
        "category": "Entertainment: Video Games",
        "correct_answer": "True",
        "difficulty": "easy",
        "incorrect_answers": Array [
          "False",
        ],
        "question": "&quot;Undertale&quot; is an RPG created by Toby Fox and released in 2015.",
        "type": "boolean",
      },
      Object {
        "category": "Science: Mathematics",
        "correct_answer": "1,000,000",
        "difficulty": "hard",
        "incorrect_answers": Array [
          "10",
          "1,000,000,000",
          "1000",
        ],
        "question": "How many zeptometres are inside one femtometre?",
        "type": "multiple",
      },
      Object {
        "category": "Entertainment: Musicals & Theatres",
        "correct_answer": "Newts",
        "difficulty": "hard",
        "incorrect_answers": Array [
          "Humans",
          "Sharks",
          "Squids",
        ],
        "question": "In Macbeth, the eyes of what animals were used in the Witches&#039; cauldron?",
        "type": "multiple",
      },
      Object {
        "category": "General Knowledge",
        "correct_answer": "Finding shapes in clouds",
        "difficulty": "hard",
        "incorrect_answers": Array [
          "Sleeping with your eyes open",
          "Breaking glass with your voice",
          "Swimming in freezing water",
        ],
        "question": "Nephelococcygia is the practice of doing what?",
        "type": "multiple",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
